tests/run-make/rustdoc-io-error/rmake.rs

```diff
@@ -0,0 +1,31 @@
+// This test verifies that rustdoc doesn't ICE when it encounters an IO error
+// while generating files. Ideally this would be a rustdoc-ui test, so we could
+// verify the error message as well.
+//
+// It operates by creating a temporary directory and modifying its
+// permissions so that it is not writable. We have to take special care to set
+// the permissions back to normal so that it's able to be deleted later.
+
+use run_make_support::{rustdoc, tmp_dir};
+use std::fs;
+
+fn main() {
+    let out_dir = tmp_dir().join("rustdoc-io-error");
+    let output = fs::create_dir(&out_dir).unwrap();
+    let mut permissions = fs::metadata(&out_dir).unwrap().permissions();
+    permissions.set_readonly(true);
+    fs::set_permissions(&out_dir, permissions.clone()).unwrap();
+
+    let output = rustdoc().input("foo.rs").output(&out_dir).command_output();
+
+    // Changing back permissions.
+    permissions.set_readonly(false);
+    fs::set_permissions(&out_dir, permissions).unwrap();
+
+    #[cfg(not(windows))]
```

Comment:
Is there a reason `assert_eq!(output.status.code().unwrap(), 1);` is non-Windows? Basic r/w permissions are available on Windows too right? Does rustdoc behave differently on Windows for a non-modifiable output directory?

Reply:
Because I couldn't ensure the exact error code value on Windows (I don't have windows) so for now I only check that it fails whatever the OS, and if it's unix, then I check the error code is 1.

Reply:
You could try editing PR CI to enable some Windows runners in PR CI and see if the error code is the same on Windows

Reply:
The original test was running fine on Windows so I suppose it's fine to just check the error code whatever the platform.